src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenLocalFunctionTests.cs

```diff
@@ -5357,14 +5357,46 @@ static void validate(ModuleSymbol module)
                 var attrs1 = localFn1.GetAttributes();
                 Assert.Equal("CompilerGeneratedAttribute", attrs1.Single().AttributeClass.Name);
 
-                // PROTOTYPE(local-function-attributes): consider preventing the lowered local function from containing DynamicAttribute
                 Assert.Equal("DynamicAttribute", localFn1.GetReturnTypeAttributes().Single().AttributeClass.Name);
 
                 var param = localFn1.Parameters.Single();
                 Assert.Equal("DynamicAttribute", param.GetAttributes().Single().AttributeClass.Name);
             }
         }
 
+        [Fact]
+        public void LocalFunctionParamsArray_NoParamArrayAttribute()
+        {
+            var source = @"
+class C
+{
+    void M()
+    {
+#pragma warning disable 8321
+        void local1(params int[] arr) { }
+    }
+}
+";
+            var verifier = CompileAndVerify(
+                source,
+                options: TestOptions.DebugDll.WithMetadataImportOptions(MetadataImportOptions.All),
+                parseOptions: TestOptions.RegularPreview,
+                symbolValidator: validate);
+
+            static void validate(ModuleSymbol module)
+            {
+                var cClass = module.GlobalNamespace.GetMember<NamedTypeSymbol>("C");
+                var localFn1 = cClass.GetMethod("<M>g__local1|0_0");
+                var attrs1 = localFn1.GetAttributes();
+                Assert.Equal("CompilerGeneratedAttribute", attrs1.Single().AttributeClass.Name);
+
+                Assert.Empty(localFn1.GetReturnTypeAttributes());
+
+                var param = localFn1.Parameters.Single();
+                Assert.Empty(param.GetAttributes());
```

Comment:
>Assert.Empty(param.GetAttributes()); [](start = 16, length = 36)

Is this intentional? Did we explicitly decided to not synthesize the attribute?

Reply:
This is pre-existing behavior for local functions with `params` parameters. I'm not sure if this happened by accident or if it was a deliberate decision.

---
In reply to: [373203102](https://github.com/dotnet/roslyn/pull/41299#discussion_r373203102) [](ancestors = 373203102)